src/components/users/UserForm.jsx

```diff
@@ -0,0 +1,159 @@
+import React, { useState, useContext, useEffect } from 'react';
+import { Formik, Form } from 'formik';
+import { useTranslation } from 'react-i18next';
+import { TextField } from '@material-ui/core';
+import { Typography } from '@mui/material';
+import { useNavigate, useLocation } from 'react-router-dom';
+import Button from '@mui/material/Button';
+import ToastContext from "../../context/ToastContext";
+import UserValidationSchema from '../../pages/users/UserValidationSchema';
+import { reset } from '../../context/redux/actions';
+import store from '../../context/redux/store';
+
+export default function UserForm() {
+    const [singleUser, setSingleUser] = useState([]);
+    const { t } = useTranslation();
+    const [isLoading, setIsLoading] = useState(false);
+    const { setToastState } = useContext(ToastContext);
+    const navigate = useNavigate();
+    const location = useLocation();
+
+    const getUserId = () => {
+        return (location.pathname).slice(7);
+    };
+
+    const userId = getUserId();
+
+    useEffect(() => {
+        fetchUser();
+    }, [userId]);
+
+    const fetchUser = async () => {
```

Comment:
fetchUser and updateUser can sit outside the component itself in order not to clutter the component. you can send them state parameters and callback parameters from this component, but their definition can sit outside of it